### PR TITLE
Fix the release mechanism

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

Fix the release mechanism that was not touched since some time and no longer working.

---

**Related issue :** #239

### CODE REVIEW

File: .github/workflows/pre-release.yml [hunk 1]
```diff
@@ -1,96 +1,110 @@
...
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+      - uses: actions/checkout@v3
+      - name: get git ready
+        run: |
+          git config user.name "egibot"
+          git config user.email "egibot@egi.eu"
+      - name: Extract version
+        id: extract-version
+        run: |
+          # this uses GNU sed -z!
```
Comment (gwarf) [inline .github/workflows/pre-release.yml:34]: I don't see any `sed -z`

File: cloud_info_provider/providers/mesos.py [hunk 2]
```diff
@@ -95,7 +95,12 @@ def _update(old, new):
         # gather & update dynamic info
         for api_endp in self.api_endpoints:
             api_url = "/".join([self.framework_url, api_endp])
-            r = requests.get(api_url, headers=self.headers, verify=self.cacert)
+            r = requests.get(
+                api_url,
+                headers=self.headers,
+                verify=self.cacert,
+                timeout=self.opts.timeout,
+            )
```
Comment (gwarf) [inline cloud_info_provider/providers/mesos.py:103]: From where come those changes?

**Review by gwarf — APPROVED** [review]
LGTM, added a few comments.

File: .github/workflows/pre-release.yml (hunk 1 above)
Comment (enolfc) [reply .github/workflows/pre-release.yml:34]:
```suggestion
```

was from previous action

File: cloud_info_provider/providers/mesos.py (hunk 2 above)
Comment (enolfc) [reply cloud_info_provider/providers/mesos.py:103]: this was because bandit was complaining about not having timeout in request, so fixing tests

**Pull request merged**